ui-tests/src/test/java/org/openmrs/reference/page/VisitNotePage.java

```diff
@@ -26,8 +26,15 @@
     private static final By NOTE = By.id("w10");
     private static final By PROVIDER = By.id("w1");
     private static final By LOCATION = By.id("w3");
-    private static final By DEL_DIAGNOSIS = By.xpath("//div[@id='display-encounter-diagnoses-container']/ul/li/span/i");
-
+    private static final By DELETE_DIAGNOSIS = By.cssSelector("#delete-button");
+    private static final By CONFORM_DELETE_DIAGNOSIS = By.cssSelector("#confirm-delete-dialog > div.dialog-content > div > button.confirm.right");
+    private static final By CANCEL_DELETE_DIAGNOSIS = By.cssSelector("#confirm-delete-dialog > div.dialog-content > div > button.cancel");
+    private static final By EDIT_VISIT_NOTE = By.cssSelector("#encountersList > li > span > i.editEncounter.edit-action.icon-pencil");
+    private static final By VIEW_VISIT_NOTE  = By.cssSelector("#encountersList > li > span > i.viewEncounter.view-action.icon-file-alt");
```

Comment:
Does visit Note have both edit and view functionalities? you could be instead landing on ones for the listed encounters

Reply:
So visit Note comprises of view button ,edit and delete however for view button, it have other similar buttons which we are not going to use by default but in actual sense those buttons can also delete and edit as well

Reply:
could you take a screenshot and share it here?

Reply:
![visitNotePage](https://user-images.githubusercontent.com/48015736/125616946-75226218-113a-4c43-aa12-2c72dd3a64a6.png)
So here is real visit Note  , however all the functionalities of visit Note and visits are in visitsPage, 
![Uploading visitsNotes.png…]()

Reply:
I think most of these CSS selectors are one class too deep, e.g. `#confirm-delete-dialog > div.dialog-content > div > button.confirm` should be sufficient. Ditto for `#encountersList > li > span > i.editEncounter.edit-action`. For edit and view, we might want to add the `:first-of-type` pseudo-selector.

Reply:
Alright, this is a little insane, but I think a better way to get the view-action or edit-action is this:

```java
By.xpath("//ul[@class=\"encounter-details\"]/child::*//i[contains(@class, \"icon-vitals\")][1]/following::*//i[contains(@class, \"view-action\")][1]");
By.xpath("//ul[@class=\"encounter-details\"]/child::*//i[contains(@class, \"icon-vitals\")][1]/following::*//i[contains(@class, \"edit-action\")][1]")
```

This attempts to select the first `.view-action` or `.edit-action` item that comes after the first encounter with a vitals icon. That way, this should work even if there are additional encounter types.

Reply:
I tried rectifying this following the same suggestions but combining them breaks it